### PR TITLE
Adding tests for Unix Domain Socket

### PR DESCRIPTION
**Pull request by EldarShalev**
Adding tests for recently Unix Domain Socket patch
* creating new `for-loop` for `test_uds_list` 
* creating new file `UDS.pm` to distinguish UDS tests from original tests 
* extends `Funclet.pm`, `Utility.pm` and `Verifier.pl` to support new tests

### CODE REVIEW

**Comment by EldarShalev** [comment]
I run the tests in my environment, this is the output: 
example


Also created `uds_test.dump` and `uds_test.log`

File: contrib/jenkins_tests/test.sh [hunk 1]
```diff
@@ -80,6 +81,14 @@ for test_link in $test_ip_list; do
...
 done
 
+for test_link in $test_uds_list; do
+	test_in="uds"
+	test_name="uds_test"
+	$timeout_exe ${WORKSPACE}/tests/verifier/verifier.pl -a ${test_app} \
+            -t ${test_in} -s ${test_link} -l ${test_dir}/${test_name}.log \
+            --progress=0
+done
+
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/test.sh:118]: Only thing you need is just 
```
test_ip_list="local:/tmp/sockperf-test-$$"
test_list="uds"

for test_link in $test_ip_list; do
	for test in $test_list; do
<the same as above>
        done
done
```

File: tests/verifier/lib/UDS.pm [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            note => '#1 - ping-pong w/o arguments',
+            pre_proc => \&te_def_pre_proc,
+            server_proc => \&te_def_server_proc,
+            server_arg => 'sr -i TARGET() ',
+            client_proc => \&te_def_client_proc,
+            client_arg => 'pp -i TARGET() ',
+            result_proc => \&te_def_result_proc,
+            result_arg => {
+                server => {
+                    success => ['Test end', 'interrupted by', 'exit'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: please consider adding unix specific output matching values

File: tests/verifier/lib/UDS.pm [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                    failure => ['Segmentation fault', 'Assertion', 'ERROR']
+                },
+                client => {
+                    success => ['Test ended', 'Summary: Latency is', 'Warmup stage \(sending a few dummy messages\)...'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: please consider adding unix specific output matching values

File: tests/verifier/lib/UDS.pm [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            note => '#2 - ping-pong option --tcp',
+            pre_proc => \&te_def_pre_proc,
+            server_proc => \&te_def_server_proc,
+            server_arg => 'sr -i TARGET() --tcp',
+            client_proc => \&te_def_client_proc,
+            client_arg => 'pp -i TARGET() --tcp',
+            result_proc => \&te_def_result_proc,
+            result_arg => {
+                server => {
+                    success => ['Test end', 'interrupted by', 'exit'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: please consider adding unix specific output matching values

File: tests/verifier/lib/UDS.pm [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                    failure => ['Segmentation fault', 'Assertion', 'ERROR']
+                },
+                client => {
+                    success => ['Test ended', 'Summary: Latency is', 'Warmup stage \(sending a few dummy messages\)...'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: please consider adding unix specific output matching values

File: tests/verifier/lib/UDS.pm [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+use TE::Common;
+use TE::Funclet;
+use TE::Utility;
+
+
+our $test_suite_uds = 
+    [
+{
+            name => 'tc1',
+            note => '#1 - ping-pong w/o arguments',
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: it should be UDP test. Is it valid for unix socket.

**Comment by igor-ivanov** [comment]
please add issue number in commit

**Comment by igor-ivanov** [comment]
please update verifier tool version as 0.3

File: tests/verifier/lib/UDS.pm (hunk 2 above)
Comment (EldarShalev) [reply tests/verifier/lib/UDS.pm]: for which part? 
the output is the same as any other protocl, it's sockperf output for success

**Comment by EldarShalev** [comment]
> please add issue number in commit

Which number?

File: contrib/jenkins_tests/test.sh (hunk 1 above)
Comment (EldarShalev) [reply contrib/jenkins_tests/test.sh:118]: `test_ip_list` and `test_list` are in use for the original flow, 
so I can create `test_uds_list` and `test_uds_path` to distinguish between UDS and IP, and as we spoke to create new for loop for UDS separately

File: tests/verifier/lib/UDS.pm (hunk 6 above)
Comment (EldarShalev) [reply tests/verifier/lib/UDS.pm]: are you referring to the name? the note? both?

File: contrib/jenkins_tests/test.sh (hunk 1 above)
Comment (EldarShalev) [reply contrib/jenkins_tests/test.sh:118]: e.g 

```
test_uds_list="uds"
test_uds_path="local:/tmp/sockperf-test-$$"


for test_link in $test_uds_list; do
	for test in $test_uds_path; do
<the same as above>
        done
done
```

File: tests/verifier/lib/UDS.pm (hunk 2 above)
Comment (igor-ivanov) [reply tests/verifier/lib/UDS.pm]: > for which part? the output is the same as any other protocl, it's sockperf output for success

You add test specific for unix socket. Sockperf prints output related unix socket. So please consider checking this output in test adding as match criteria.

File: tests/verifier/lib/UDS.pm (hunk 6 above)
Comment (igor-ivanov) [reply tests/verifier/lib/UDS.pm]: Is this test is Linux specific or not?

**Comment by igor-ivanov** [comment]
> > please add issue number in commit
> 
> Which number?

You can use [2935075] or create new ticket

File: tests/verifier/lib/UDS.pm (hunk 6 above)
Comment (EldarShalev) [reply tests/verifier/lib/UDS.pm]: Some test are for Linux and some works only for Windows 
e.g feedfile path is different (linux is /tmp/test and windows is different with c:\tmp\test)
another example is AF_UNIX, only work at tcp mode in windows while in linux it works tcp and udp

Comment (igor-ivanov) [reply tests/verifier/lib/UDS.pm]: So, udp test is Linux specific, correct and launching verifier under Windows should fail.

Comment (EldarShalev) [reply tests/verifier/lib/UDS.pm]: Yes, but this test run also for windows setup or we have different tests for windows? 
if it's the same- how can I distinguish between the environments?

File: contrib/jenkins_tests/test.sh (hunk 1 above)
Comment (igor-ivanov) [reply contrib/jenkins_tests/test.sh:118]: > `test_ip_list` and `test_list` are in use for the original flow, so I can create `test_uds_list` and `test_uds_path` to distinguish between UDS and IP, and as we spoke to create new for loop for UDS separately

Just reassign these variables

File: contrib/jenkins_tests/test.sh [hunk 7]
```diff
@@ -80,6 +79,20 @@ for test_link in $test_ip_list; do
...
+for test_link in $test_ip_list; do
+	for test in $test_list; do
+		test_in="uds"
+		test_name="uds_test"
+		$timeout_exe ${WORKSPACE}/tests/verifier/verifier.pl -a ${test_app} \
+				-t ${test_in} -s ${test_link} -l ${test_dir}/${test_name}.log \
+				--progress=0
+	done
+done
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/test.sh:117]: Could you please follow recommendation https://github.com/Mellanox/sockperf/pull/200#discussion_r956244189
It allows to do many things that are required in single style. These useful actions are missed in your last change (for example tap file output)

File: tests/verifier/lib/UDS.pm [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+            note => '#1 - ping-pong w/o arguments (default is udp)',
+            pre_proc => \&te_def_pre_proc,
+            server_proc => \&te_def_server_proc,
+            server_arg => 'sr -i TARGET() ',
+            client_proc => \&te_def_client_proc,
+            client_arg => 'pp -i TARGET() ',
+            result_proc => \&te_def_result_proc,
+            result_arg => {
+                server => {
+                    success => ['Test end', 'interrupted by', 'exit', 'UDS type is SOCK_DGRAM'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: "UDS type ..." looks as invalid according last master and https://github.com/Mellanox/sockperf/pull/193/files#diff-442cdda1ab1d725b607a7d8100d86dafbf8148ef15d49c3406a9a776d1ab821bR41

File: tests/verifier/lib/UDS.pm [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+                    failure => ['Segmentation fault', 'Assertion', 'ERROR']
+                },
+                client => {
+                    success => ['Test ended', 'Summary: Latency is', 'Warmup stage \(sending a few dummy messages\)...', 'UDS type is SOCK_DGRAM'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: "UDS type ..." looks as invalid according last master and https://github.com/Mellanox/sockperf/pull/193/files#diff-442cdda1ab1d725b607a7d8100d86dafbf8148ef15d49c3406a9a776d1ab821bR41

File: tests/verifier/lib/UDS.pm [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+            client_proc => \&te_def_client_proc,
+            client_arg => 'pp -i TARGET() --tcp',
+            result_proc => \&te_def_result_proc,
+            result_arg => {
+                server => {
+                    success => ['Test end', 'interrupted by', 'exit', 'UDS type is SOCK_STREAM'],
+                    failure => ['Segmentation fault', 'Assertion', 'ERROR']
+                },
+                client => {
+                    success => ['Test ended', 'Summary: Latency is', 'Warmup stage \(sending a few dummy messages\)...', 'UDS type is SOCK_STREAM'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: "UDS type ..." looks as invalid according last master and https://github.com/Mellanox/sockperf/pull/193/files#diff-442cdda1ab1d725b607a7d8100d86dafbf8148ef15d49c3406a9a776d1ab821bR41

File: tests/verifier/lib/UDS.pm [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+            client_proc => \&te_def_client_proc,
+            client_arg => 'pp -i TARGET() --tcp -b10',
+            result_proc => \&te_def_result_proc,
+            result_arg => {
+                server => {
+                    success => ['Test end', 'interrupted by', 'exit', 'UDS type is SOCK_STREAM'],
+                    failure => ['Segmentation fault', 'Assertion', 'ERROR']
+                },
+                client => {
+                    success => ['Test ended', 'Summary: Latency of burst of 10 messages', 'UDS type is SOCK_STREAM'],
```
Comment (igor-ivanov) [inline tests/verifier/lib/UDS.pm]: "UDS type ..." looks as invalid according last master and https://github.com/Mellanox/sockperf/pull/193/files#diff-442cdda1ab1d725b607a7d8100d86dafbf8148ef15d49c3406a9a776d1ab821bR41

File: tests/verifier/lib/UDS.pm (hunk 8 above)
Comment (EldarShalev) [reply tests/verifier/lib/UDS.pm]: you right, 
I am not sure what else I can bring into "success" part which is related to UDS only

Comment (EldarShalev) [reply tests/verifier/lib/UDS.pm]: Any suggestions?

File: tests/verifier/lib/UDS.pm (hunk 9 above)
Comment (EldarShalev) [reply tests/verifier/lib/UDS.pm]: you right I will remove "UDS type.."

**Comment by vasily-v-ryabov** [comment]
bot:retest

**Comment by jeniaka** [comment]
bot:retest

**Comment by EldarShalev** [comment]
> bot:retest

I see one tests that was failed (except multicast of course)
`FAIL   tcp-ul       tc7          #7 - under-load option -t30`

Any idea why?

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by vasily-v-ryabov** [comment]
bot:retest

**Comment by vasily-v-ryabov** [comment]
bot:retest

**Pull request merged**